src/main/scala/com/microsoft/azure/cosmosdb/spark/CosmosDBConnection.scala

```diff
@@ -213,25 +214,33 @@ private[spark] case class CosmosDBConnection(config: Config) extends CosmosDBLog
   }
 
   def getCollectionThroughput: Int = {
-    var offers = documentClient.queryOffers(s"SELECT * FROM c where c.offerResourceId = '${getCollection.getResourceId}'", null).getQueryIterable.toList
-    if (offers.isEmpty) {
-      offers = documentClient.queryOffers(s"SELECT * FROM c where c.offerResourceId = '${getDatabase.getResourceId}'", null).getQueryIterable.toList
-      // database throughput
-      if (offers.isEmpty) {
-          throw new IllegalStateException("Cannot find the collection corresponding offer.")
-      }
-    }
+    collectionThroughput match {
+      case Some(value) => value
+      case None => {
+        var offers = documentClient.queryOffers(s"SELECT * FROM c where c.offerResourceId = '${getCollection.getResourceId}'", null).getQueryIterable.toList
```

Comment:
ideally we should be using SqlQuerySpec to prevent sql injection.

However this not introduced in this PR, so out of scope of the PR. for future spark v3 version we should try to fix these.